node-js-basics-101/node_modules/uws/uws.js

```diff
@@ -0,0 +1,563 @@
+            this.onpong = f;
+        } else if (eventName === 'open') {
+            if (this.internalOnOpen !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnOpen = f;
+        } else if (eventName === 'error' && this instanceof WebSocketClient) {
+            if (this.internalOnError !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnError = f;
+        }
+        return this;
+    }
+
+    once(eventName, f) {
+        if (eventName === 'message') {
+            if (this.internalOnMessage !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnMessage = (message) => {
+                this.internalOnMessage = noop;
+                f(message);
+            };
+        } else if (eventName === 'close') {
+            if (this.internalOnClose !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.internalOnClose = (code, message) => {
+                this.internalOnClose = noop;
+                f(code, message);
+            };
+        } else if (eventName === 'ping') {
+            if (this.onping !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.onping = () => {
+                this.onping = noop;
+                f();
+            };
+        } else if (eventName === 'pong') {
+            if (this.onpong !== noop) {
+                throw Error(EE_ERROR);
+            }
+            this.onpong = () => {
+                this.onpong = noop;
+                f();
+            };
+        }
+        return this;
+    }
+
+    removeAllListeners(eventName) {
+        if (!eventName || eventName === 'message') {
+            this.internalOnMessage = noop;
+        }
+        if (!eventName || eventName === 'close') {
+            this.internalOnClose = noop;
+        }
+        if (!eventName || eventName === 'ping') {
+            this.onping = noop;
+        }
+        if (!eventName || eventName === 'pong') {
+            this.onpong = noop;
+        }
+        return this;
+    }
+
+    removeListener(eventName, cb) {
+        if (eventName === 'message' && this.internalOnMessage === cb) {
+            this.internalOnMessage = noop;
+        } else if (eventName === 'close' && this.internalOnClose === cb) {
+            this.internalOnClose = noop;
+        } else if (eventName === 'ping' && this.onping === cb) {
+            this.onping = noop;
+        } else if (eventName === 'pong' && this.onpong === cb) {
+            this.onpong = noop;
+        }
+        return this;
+    }
+
+    get OPEN() {
+        return WebSocketClient.OPEN;
+    }
+
+    get CLOSED() {
+        return WebSocketClient.CLOSED;
+    }
+
+    get readyState() {
+        return this.external ? WebSocketClient.OPEN : WebSocketClient.CLOSED;
+    }
+
+    get _socket() {
+        const address = this.external ? native.getAddress(this.external) : new Array(3);
```

Comment:
__[jshint]__
* 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

<!-- Comment by [SideCI](https://sideci.com) -->